### ISSUE TITLE
Fontify org-roam links that point to nowhere

### ISSUE THREAD

**Issue by Wetlize** [jethrokuan/org-roam#555]
### Brief Abstract
Since the original location of org-roam links can change and the origin itself can potentially not even exist (in which case following the link would create an empty buffer), I think it would be useful to apply a separate face for org-roam links that point to nowhere.

### Extra information
This was stimulated by #554, when I needed to see an immediate feedback for validation of my links.

Also, if #548 ever end-up to be implemented, then the implementation would also require validation of headings in order to not confuse users with false-negative passing (unless we restrict the semantic of this fontification purely to file paths). 

### Proposed Implementation (if any)
As a reference and proof of concept, this is what I've quickly hacked in my config to achieve this:

```elisp
(defface my-org-roam-link-empty
  '((t (:inherit (org-warning org-link))))
  "Face for Org-roam links pointing to files that don't exist."
  :group 'org-roam-faces)

(defun my-org-roam--highlight-empty-links-a (orig-fn path)
  "Fontify org-roam file links that point to nowhere."
  (let ((face (funcall orig-fn path)))
    (if (and (eq face 'org-roam-link)
             (not (file-exists-p path)))
        'my-org-roam-link-empty
      face)))

(advice-add #'org-roam--roam-link-face :around
            #'my-org-roam--highlight-empty-links-a)
```

### Please check the following:
- [x] No similar feature requests

**Comment by zaeph** [comment]
Arguably, this is something that should be done upstream, but since we have the infrastructure in place to validate links here, we might as well implement it in Org-roam.

**Comment by jethrokuan** [comment]
Closed with #560
